src/libsync/account.cpp

```diff
@@ -302,37 +302,37 @@ void Account::setCapabilities(const Capabilities &caps)
     _capabilities = caps;
 }
 
-QString Account::serverVersion() const
+QString Account::serverProduct() const
 {
-    return _serverVersion;
+    return _serverProduct;
 }
 
-int Account::serverVersionInt() const
+QString Account::serverVersionString() const
 {
-    // FIXME: Use Qt 5.5 QVersionNumber
-    auto components = serverVersion().split(QLatin1Char('.'));
-    return makeServerVersion(components.value(0).toInt(),
-        components.value(1).toInt(),
-        components.value(2).toInt());
-}
-
-int Account::makeServerVersion(int majorVersion, int minorVersion, int patchVersion)
-{
-    return (majorVersion << 16) + (minorVersion << 8) + patchVersion;
+    return _serverVersion;
 }
 
 bool Account::serverVersionUnsupported() const
 {
-    if (serverVersionInt() == 0) {
+    if (serverVersion().isNull()) {
         // not detected yet, assume it is fine.
         return false;
     }
+    if (serverProduct() == QLatin1String("Infinite Scale")) {
+        return false;
```

Comment:
We should probably leave a TODO/FIXME here, since this is a workaround. In the future, such checks may be required again.